services/QuillLMS/client/app/bundles/Teacher/components/progress_reports/activity_analysis/activity_pack.tsx

```diff
@@ -0,0 +1,201 @@
+import * as React from 'react';
+import Pluralize from 'pluralize';
+
+import activityFromClassificationId from '../../modules/activity_from_classification_id.js';
+import useScaledTableColumnWidths from '../../../../Shared/hooks/useScaledTableColumnWidths';
+import { DataTable, Tooltip, EXTRA_SMALL_ICON_BASE_SRC } from '../../../../Shared/index';
+import { nonRelevantActivityClassificationIds } from '../../../../../modules/activity_classifications';
+import { requestGet } from '../../../../../modules/request';
+
+const chevronRightImgSrc = `${EXTRA_SMALL_ICON_BASE_SRC}/chevron-right.svg`;
+
+interface Classroom {
+  assignedStudentCount: number;
+  name: string;
+  totalStudentCount: number;
+}
+
+interface ClassroomActivity {
+  activityClassificationId: number;
+  activityId: number;
+  completedCount: number;
+  cumulativeScore: number;
+  name: string;
+}
+
+interface ActivityPackProps {
+  data: {
+    unitId: number;
+    unitName: string;
+    classrooms: Classroom[];
+    classroomActivities: Map<number, ClassroomActivity>;
+  };
+}
+
+const ActivityPack: React.FC<ActivityPackProps> = ({ data }) => {
+  const containerRef = React.useRef<HTMLElement>(null);
+
+  const columnWidths = useScaledTableColumnWidths({
+    additionalWidthRestrictions: 2,
+    columnWidths: [
+      { minWidth: 500, width: 720 },
+      { minWidth: 150, width: 230 },
+      { minWidth: 100, width: 230 },
+    ],
+    containerRef,
+    maxTableWidth: 1198,
+    minTableWidth: 750,
+  });
+
+  const headers = [
+    {
+      attribute: 'tool',
+      name: 'Tool',
+      renderCellsAsHeaders: true,
+      width: columnWidths[0],
+    },
+    {
+      attribute: 'completion',
+      name: 'Completion',
+      width: columnWidths[1],
+    },
+    {
+      attribute: 'averageScore',
+      name: 'Average score',
+      rowSectionClassName: 'average-score',
+      width: columnWidths[2],
+    },
+  ];
+
+  const { unitName, classroomActivities, unitId, classrooms } = data;
+
+  const numberOfStudentsAssignedToUnit = () =>
+    classrooms.reduce((sum, c) => sum + c.assignedStudentCount, 0);
+
+  const handleReportLinkClick = (unitId: number, activityId: number) => {
+    requestGet(
+      `/teachers/progress_reports/report_from_unit_and_activity/u/${unitId}/a/${activityId}`,
+      (data) => {
+        window.location.href = data.url;
+      },
+      () =>
+        alert(
+          'This report is not yet available. Please check back when at least one of your students has completed this activity.'
+        )
+    );
+  };
+
+  const renderAssignedToSection = () => (
+    <div className="assigned-to">
+      <span>
+        Assigned to {classrooms.length} {Pluralize('class', classrooms.length)}:
+      </span>
+      <ul>
+        {classrooms.map((c, i) => (
+          <li key={i}>
+            <b>{c.name}</b>
+            <span>
+              ({c.assignedStudentCount}/{c.totalStudentCount}{' '}
```

Comment:
This will print `Infinity` if totalStudentCount is zero.

Reply:
@happythenewsad It actually isn't doing the calculation here, it's just rendering the counts as a string with the `/` in between (it would be doing the calculation if it were `{c.assignedStudentCount/c.totalStudentCount}`. So it would render `0/0`, which is admittedly weird, but that's not a state the data should be in anyway.